addons/binding/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/discovery/HDPowerViewHubDiscoveryService.java

```diff
@@ -26,33 +29,53 @@
  */
 public class HDPowerViewHubDiscoveryService extends AbstractDiscoveryService {
 
+    private final Runnable scanner;
+    private ScheduledFuture<?> backgroundFuture;
+
     public HDPowerViewHubDiscoveryService() {
-        super(HDPowerViewBindingConstants.SUPPORTED_THING_TYPES_UIDS, 600, true);
+        super(Collections.singleton(HDPowerViewBindingConstants.THING_TYPE_HUB), 600, true);
+        scanner = createScanner();
     }
 
     @Override
     protected void startScan() {
-        try {
-            NbtAddress address = NbtAddress.getByName(HDPowerViewBindingConstants.NETBIOS_NAME);
-            if (address != null) {
-                String ip = address.getInetAddress().getHostAddress();
-                ThingUID thingUID = new ThingUID(HDPowerViewBindingConstants.THING_TYPE_HUB, ip.replace('.', '_'));
-                DiscoveryResult result = DiscoveryResultBuilder.create(thingUID)
-                        .withProperty(HDPowerViewHubConfiguration.IP_ADDRESS, ip)
-                        .withLabel("PowerView Hub (" + ip + ")").build();
-                thingDiscovered(result);
-                stopScan();
-            } else {
-                stopScan();
-            }
-        } catch (UnknownHostException e) {
-            stopScan();
-        }
+        scheduler.execute(scanner);
     }
 
     @Override
     protected void startBackgroundDiscovery() {
-        startScan();
+        if (backgroundFuture != null && !backgroundFuture.isDone()) {
+            backgroundFuture.cancel(true);
+            backgroundFuture = null;
+        }
+        backgroundFuture = scheduler.scheduleAtFixedRate(scanner, 0, 60, TimeUnit.SECONDS);
```

Comment:
scheduleWithFixedDelay is usually better, since it avoids jobs piling up (e.g. while being in a breakpoint while debugging).